pkg/controller/dns/mock_dns_config.go

```diff
@@ -0,0 +1,116 @@
+package dns
+
+import (
+	"github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type mockDnsConfig struct {
```

Comment:
i don't understand what the point of the mock config is in the first place. this isn't something you'd normally mock